src/Sylius/Bundle/CoreBundle/EventListener/CatalogPromotionEventListener.php

```diff
@@ -40,21 +41,45 @@ public function __construct(
         $this->dateTimeProvider = $dateTimeProvider;
     }
 
+    public function dispatchCatalogPromotionCreatedEvent(GenericEvent $event): void
+    {
+        /** @var CatalogPromotionInterface $catalogPromotion */
+        $catalogPromotion = $event->getSubject();
+        Assert::isInstanceOf($catalogPromotion, CatalogPromotionInterface::class);
+
+        $this->eventBus->dispatch(
+            new CatalogPromotionCreated($catalogPromotion->getCode()),
+            $this->calculateStartDateStamp($catalogPromotion)
+        );
+
+        $this->dispatchCatalogPromotionEndedEvent($catalogPromotion);
```

Comment:
I'm wondering if in the next iteration we couldn't extract this logic and reuse it here and in Api Platform context. Same for update action